packages/prisma/schema.prisma

```diff
@@ -320,6 +320,35 @@ model Team {
   @@unique([slug, parentId])
 }
 
+model AwaitingPaymentTeam {
+  id                  Int      @id @default(autoincrement())
+  /// @zod.min(1)
+  name                String
+  /// @zod.min(1)
+  slug                String?
+  logo                String?
+  logoUrl             String?
+  calVideoLogo        String?
+  appLogo             String?
+  appIconLogo         String?
+  bio                 String?
+  hideBranding        Boolean  @default(false)
+  isPrivate           Boolean  @default(false)
+  hideBookATeamMember Boolean  @default(false)
+  createdAt           DateTime @default(now())
+  /// @zod.custom(imports.teamMetadataSchema)
+  metadata            Json?
+  theme               String?
+  brandColor          String?
+  darkBrandColor      String?
+  parentId            Int?
+  timeFormat          Int?
+  timeZone            String   @default("Europe/London")
+  weekStart           String   @default("Sunday")
+
+  @@unique([slug, parentId])
+}
```

Comment:
Can we achieve this without having to repeat this code? If we make changes to Team we need to double check here as well. I would rather have some kind of "pendingPayment" Boolean value and filter out on team queries. We could even use a prisma middleware to omit teams with "pendingPayment" by default.

Reply:
We can follow the soft-delete approach for the Prisma Docs https://www.prisma.io/docs/orm/prisma-client/client-extensions/middleware/soft-delete-middleware#option-2-use-middleware-to-determine-the-behavior-of-readupdate-queries-for-deleted-records

Reply:
Great idea! Added "pendingPayment" to schema and refactored existing code to work with it in commit [refactor: replace AwaitingPaymentTeam table with Team.pendingPayment](https://github.com/calcom/cal.com/pull/13211/commits/a2f25c1c3073fdf8375e5a2508616bc8f7196689).

Then, added prisma filtering extension in commit [feat: filter out teams with pending payments](https://github.com/calcom/cal.com/pull/13211/commits/9f19a5bff43da659082936f456fddc250a562a1c).

In the handler that is invoked after successful stripe checkout "apps/web/pages/api/teams/api/create.ts" I tested "prisma.team.findFirst({where: {id: pendingPaymentTeamId}})" aka team we created in POST "/teams" but with "pendingPayment: true" and null was returned, so the filter works. Notably, the filter only works for find operations and not update, because we invoke update after successful checkout to set "pendingPayment: false".